packages/gestalt/src/ScrollableContainer.js

```diff
@@ -28,31 +37,61 @@ type Props = {|
   overflow?: ScrollableOverflow,
 |};
 
-function ScrollableContainer({ children, height = '100%', overflow = 'auto' }: Props): Node {
+type InternalProps = {|
+  children?: Node,
+  height?: Dimension,
+  onScroll?: () => void,
+  overflow?: ScrollableOverflow,
+  padding?: Padding,
+|};
+
+// ScrollableContainerWithForwardRef is the ScrollableContainer to be used internally, within scrollable components (e. Modal, Sheet).
+// It has an extended API with private props (onScroll, padding, and ref) to maintain border shadows in the component main content container.
+const ScrollableContainerWithForwardRef: AbstractComponent<
+  InternalProps,
+  HTMLDivElement,
+> = forwardRef<InternalProps, HTMLDivElement>(function ScrollableContainer(
+  { children, onScroll, padding, height = '100%', overflow = 'auto' },
+  ref,
+): Node {
   const { addRef } = useScrollableContainer();
-  const anchorRef = useRef<?HTMLDivElement>(null);
+  const anchorRef = useRef<HTMLDivElement | null>(null);
+  // When using both forwardRef and innerRef, React.useimperativehandle() allows a parent component
```

Comment:
Thanks for the comment, very helpful 👍

Reply:
I believe all 
`React.useimperativehandle` uses have this comment to explicit their use. This trick was courtesy of jack hsu (I believe)